roslibrust/src/integration_tests.rs

```diff
@@ -205,4 +230,42 @@ mod integration_tests {
         }
         Ok(())
     }
+
+    #[tokio::test]
+    async fn self_service_call() -> TestResult {
+        let _ = simple_logger::SimpleLogger::new()
+            .with_level(log::LevelFilter::Trace)
+            .without_timestamps()
+            .init();
+
+        let opt = ClientOptions::new(LOCAL_WS).timeout(TIMEOUT);
+        let mut client = Client::new_with_options(opt).await?;
+
+        let cb =
+            |_req: SetBoolRequest| -> Result<SetBoolResponse, Box<dyn std::error::Error + Send + Sync>> {
+                Ok(SetBoolResponse {
+                    success: true,
+                    message: "call_success".to_string(),
+                })
+            };
+
+        let topic = "/self_service_call";
+
+        client
```

Comment:
Would it make sense for the API to return a handle such that the drop of said handle would unadvertise the service like in roscpp?

Reply:
https://github.com/Carter12s/roslibrust/issues/29